src/languages/ruby.js

```diff
@@ -122,56 +121,128 @@ export default function(hljs) {
     end: /\}/,
     keywords: RUBY_KEYWORDS
   };
-  const STRING = {
+
+  function string_variants(prefix, delimiters) {
+    return delimiters.map((d) => {
+      return {
+        begin: regex.concat(prefix, regex.escape(d.charAt(0))),
+        end: regex.escape(d.charAt(1))
+      }
+    })
+  }
+
+  const STRING_DELIMITERS = [
+    "()",
+    "[]",
+    "{}",
+    "<>",
+    "\\/\\/",
```

Comment:
Did you mean

```suggestion
    "//",
```

`d.charAt(1)` won't work because it's 4 characters long.

```js
> d = "\\/\\/"
'\\/\\/'
> d.charAt(0)
'\\'
> d.charAt(1)
'/'
```

https://docs.ruby-lang.org/en/3.1/syntax/literals_rdoc.html#label-Percent+Literals says

> Each of the literals in described in this section may use these paired delimiters
> - [...] 
> - Any other character, as both beginning and ending delimiters.

Which seems like something regex can't handle.